pkg/synopsysctl/CRDPrint.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright (C) 2019 Synopsys, Inc.
+
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements. See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership. The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied. See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package synopsysctl
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/yaml"
+
+	alertapi "github.com/blackducksoftware/synopsys-operator/pkg/api/alert/v1"
+	blackduckapi "github.com/blackducksoftware/synopsys-operator/pkg/api/blackduck/v1"
+	opssightapi "github.com/blackducksoftware/synopsys-operator/pkg/api/opssight/v1"
+	"github.com/blackducksoftware/synopsys-operator/pkg/apps"
+	"github.com/blackducksoftware/synopsys-operator/pkg/opssight"
+	"github.com/blackducksoftware/synopsys-operator/pkg/protoform"
+	"github.com/blackducksoftware/synopsys-operator/pkg/soperator"
+)
+
+// PrintFormat represents the format to print the struct
+type PrintFormat string
+
+// Constants for the PrintFormats
+const (
+	JSON PrintFormat = "JSON"
+	YAML PrintFormat = "YAML"
+)
+
+// PrintResource prints a Resource as yaml or json. printKubeComponents allows printing the kuberentes
+// resources instead
+func PrintResource(crd interface{}, format string, printKubeComponents bool) error {
+	// print the CRD
+	if !printKubeComponents {
+		err := PrintComponents([]interface{}{crd}, format)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// print the Kube Components
+	pc := &protoform.Config{}
+	pc.SelfSetDefaults()
+	pc.DryRun = true
+	rc := &rest.Config{}
+	app := apps.NewApp(pc, rc)
+
+	operator, ok := crd.(soperator.SpecConfig)
+	if ok {
+		cList, err := operator.GetComponents()
+		if err != nil {
+			return fmt.Errorf("failed to get components: %s", err)
+		}
```

Comment:
Only think you really need to do that is different is generate the component list.  The printing of the component list can be common after it is created in the if/switch logic.

Reply:
Good Point